src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -477,6 +477,20 @@ object PruneDeadFields {
         memoizeMatrixIR(child, dep, memo)
       case MatrixUnionRows(children) =>
         children.foreach(memoizeMatrixIR(_, requestedType, memo))
+      case UnlocalizeEntries(rowsEntries, cols, fieldName) =>
+        val m = Map(MatrixType.entriesIdentifier -> fieldName)
+        val minRowEntries = minimal(rowsEntries.typ)
+        val rowsEntriesDep = minRowEntries.copy(
+          globalType = requestedType.globalType,
+          rowType = unify(rowsEntries.typ.rowType, minRowEntries.rowType, requestedType.rvRowType.rename(m))
```

Comment:
shouldn't need `minRowEntries.rowType` here. The key will necessarily stay in `requestedType.rvRowType`